### PR TITLE
feat(cli): group commands in help message

### PR DESCRIPTION
**Pull request by schoren**
This PR improves the CLI help message by grouping available commands by category. New help message looks like this:

Screenshot 2023-03-16 at 12 28 01

The order of the commands is still in alphabetical order, I didn't find an easy way to manually sort the commands. We can improve this in a follow-up PR.

### CODE REVIEW

File: cli/actions/resource.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ type ListArgs struct {
 }
 
 type ResourceActions interface {
+	Name() string
```
Comment (schoren) [inline cli/actions/resource.go:24]: unrelated code cleanup

File: cli/actions/resource.go [hunk 2]
```diff
@@ -35,28 +36,23 @@ type resourceArgs struct {
 }
 
 type ResourceArgsOption = func(args *resourceArgs)
-type ResourceRegistry map[SupportedResources]ResourceActions
-type SupportedResources string
+type ResourceRegistry map[string]ResourceActions
```
Comment (schoren) [inline cli/actions/resource.go:39]: unrelated code cleanup

File: cli/actions/resource.go [hunk 3]
```diff
@@ -35,28 +36,23 @@ type resourceArgs struct {
...
-	SupportedResourcePollingProfile SupportedResources = "pollingprofile"
-	SupportedResourceDemo           SupportedResources = "demo"
-
 	ErrResourceNotRegistered      = errors.New("resource not registered")
 	ErrNotSupportedResourceAction = errors.New("the specified resource type doesn't support the action")
 )
 
 func NewResourceRegistry() ResourceRegistry {
-	return map[SupportedResources]ResourceActions{}
+	return ResourceRegistry{}
```
Comment (schoren) [inline cli/actions/resource.go:47]: unrelated code cleanup

File: cli/actions/resource.go [hunk 4]
```diff
@@ -35,28 +36,23 @@ type resourceArgs struct {
...
 }
 
-func (r ResourceRegistry) Register(resource SupportedResources, actions ResourceActions) {
-	r[resource] = actions
+func (r ResourceRegistry) Register(actions ResourceActions) {
```
Comment (schoren) [inline cli/actions/resource.go:50]: unrelated code cleanup

File: cli/actions/resource.go [hunk 5]
```diff
@@ -35,28 +36,23 @@ type resourceArgs struct {
...
+	r[actions.Name()] = actions
 }
 
-func (r ResourceRegistry) Get(resource SupportedResources) (ResourceActions, error) {
-	resourceActions, found := r[resource]
+func (r ResourceRegistry) Get(name string) (ResourceActions, error) {
```
Comment (schoren) [inline cli/actions/resource.go:54]: unrelated code cleanup

File: cli/cmd/apply_cmd.go [hunk 6]
```diff
@@ -28,7 +29,7 @@ var applyCmd = &cobra.Command{
 			resourceType: resourceType,
 		})
 
-		resourceActions, err := resourceRegistry.Get(actions.SupportedResources(resourceType))
+		resourceActions, err := resourceRegistry.Get(resourceType)
```
Comment (schoren) [inline cli/cmd/apply_cmd.go:32]: unrelated code cleanup

File: cli/cmd/config.go [hunk 7]
```diff
@@ -48,15 +48,15 @@ func setupCommand(options ...setupOption) func(cmd *cobra.Command, args []string
 
 		configOptions := append(baseOptions, actions.WithClient(utils.GetResourceAPIClient("config", cliConfig)))
 		configActions := actions.NewConfigActions(configOptions...)
-		resourceRegistry.Register(actions.SupportedResourceConfig, configActions)
+		resourceRegistry.Register(configActions)
```
Comment (schoren) [inline cli/cmd/config.go:51]: unrelated code cleanup

File: cli/cmd/config.go [hunk 8]
```diff
@@ -48,15 +48,15 @@ func setupCommand(options ...setupOption) func(cmd *cobra.Command, args []string
...
 
 		pollingOptions := append(baseOptions, actions.WithClient(utils.GetResourceAPIClient("pollingprofile", cliConfig)))
 		pollingActions := actions.NewPollingActions(pollingOptions...)
-		resourceRegistry.Register(actions.SupportedResourcePollingProfile, pollingActions)
+		resourceRegistry.Register(pollingActions)
```
Comment (schoren) [inline cli/cmd/config.go:55]: unrelated code cleanup

File: cli/cmd/config.go [hunk 9]
```diff
@@ -48,15 +48,15 @@ func setupCommand(options ...setupOption) func(cmd *cobra.Command, args []string
...
 
 		demoOptions := append(baseOptions, actions.WithClient(utils.GetResourceAPIClient("demo", cliConfig)))
 		demoActions := actions.NewDemoActions(demoOptions...)
-		resourceRegistry.Register(actions.SupportedResourceDemo, demoActions)
+		resourceRegistry.Register(demoActions)
```
Comment (schoren) [inline cli/cmd/config.go:59]: unrelated code cleanup

File: cli/cmd/delete_cmd.go [hunk 10]
```diff
@@ -28,7 +28,7 @@ var deleteCmd = &cobra.Command{
 			resourceType: resourceType,
 		})
 
-		resourceActions, err := resourceRegistry.Get(actions.SupportedResources(resourceType))
+		resourceActions, err := resourceRegistry.Get(resourceType)
```
Comment (schoren) [inline cli/cmd/delete_cmd.go:31]: unrelated code cleanup

File: cli/cmd/export_cmd.go [hunk 11]
```diff
@@ -31,7 +31,7 @@ var exportCmd = &cobra.Command{
 			resourceType: resourceType,
 		})
 
-		resourceActions, err := resourceRegistry.Get(actions.SupportedResources(resourceType))
+		resourceActions, err := resourceRegistry.Get(resourceType)
```
Comment (schoren) [inline cli/cmd/export_cmd.go:34]: unrelated code cleanup

File: cli/cmd/get_cmd.go [hunk 12]
```diff
@@ -27,7 +27,7 @@ var getCmd = &cobra.Command{
 			resourceType: resourceType,
 		})
 
-		resourceActions, err := resourceRegistry.Get(actions.SupportedResources(resourceType))
+		resourceActions, err := resourceRegistry.Get(resourceType)
```
Comment (schoren) [inline cli/cmd/get_cmd.go:30]: unrelated code cleanup

File: cli/cmd/list_cmd.go [hunk 13]
```diff
@@ -32,7 +33,7 @@ var listCmd = &cobra.Command{
 			resourceType: resourceType,
 		})
 
-		resourceActions, err := resourceRegistry.Get(actions.SupportedResources(resourceType))
+		resourceActions, err := resourceRegistry.Get(resourceType)
```
Comment (schoren) [inline cli/cmd/list_cmd.go:36]: unrelated code cleanup

File: cli/go.mod [hunk 14]
```diff
@@ -10,11 +10,12 @@ require (
...
 	github.com/joho/godotenv v1.3.0
 	github.com/kubeshop/tracetest/server v0.0.0-20230208220354-63c9594b2160
+	github.com/mitchellh/mapstructure v1.5.0
 	github.com/pterm/pterm v0.12.55
 	github.com/segmentio/analytics-go/v3 v3.2.1
-	github.com/spf13/cobra v1.5.0
-	github.com/spf13/viper v1.13.0
-	github.com/stretchr/testify v1.8.0
+	github.com/spf13/cobra v1.6.1
+	github.com/spf13/viper v1.15.0
```
Comment (schoren) [inline cli/go.mod:17]: update deps

File: server/go.mod [hunk 15]
```diff
@@ -24,16 +24,16 @@ require (
...
 	github.com/opensearch-project/opensearch-go v1.1.0
 	github.com/orlangure/gnomock v0.20.0
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/prometheus v1.8.2-0.20211217191541-41f1a8125e66
 	github.com/segmentio/analytics-go/v3 v3.2.1
-	github.com/spf13/cobra v1.3.0
+	github.com/spf13/cobra v1.6.1
 	github.com/spf13/pflag v1.0.5
-	github.com/spf13/viper v1.10.0
+	github.com/spf13/viper v1.15.0
```
Comment (schoren) [inline server/go.mod:36]: update deps to match CLI

**Review by xoscar — APPROVED** [review]
LGTM
